components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/AddonCollectionProvider.kt

```diff
@@ -257,16 +299,18 @@ enum class SortOption(val value: String) {
     DATE_ADDED_DESC("-added")
 }
 
-internal fun JSONObject.getAddons(): List<Addon> {
+internal fun JSONObject.getAddons(language: String? = null): List<Addon> {
     val addonsJson = getJSONArray("results")
     return (0 until addonsJson.length()).map { index ->
-        addonsJson.getJSONObject(index).toAddons()
+        addonsJson.getJSONObject(index).toAddons(language)
     }
 }
 
-internal fun JSONObject.toAddons(): Addon {
+internal fun JSONObject.toAddons(language: String? = null): Addon {
     return with(getJSONObject("addon")) {
         val download = getDownload()
+        val summary = getSafeTranslations("summary", language)
+        val isLanguageInTranslations = summary.containsKey(language)
```

Comment:
@csadilek I added this extra validation to verify that the language that we are going to use as `defaultLocale` is on the actual translations, see line 330 to 334.

Reply:
Do we need to handle casing here? Should we convert everything to lower case?